### ISSUE TITLE
Valid XML gives parsing error

### ISSUE THREAD

**Issue by e-oz** [serde-rs/xml#27]
This XML is valid:
https://gist.github.com/e-oz/cbae3b90c32735e50fcd280ccab9c12a

But when I'm trying to call "from_string" to deserialize it (not to struct, just deserialize to Element), I'm getting `SyntaxError(error during lexing: "MixedElementsAndText", 1, 43)`.

If there is some workaround - please let me know.

**Comment by oli-obk** [comment]
the xml looks good to me, I'll see what I can do

**Comment by oli-obk** [comment]
I can't reproduce:

``` rust

#[test]
fn eoz() {
    let s = r#####"<?xml version="1.0" encoding="utf-8"?><Report xsi:schemaLocation="HISCI_x0020_Best_x0020_Practice_x0020_Roster http://htnawprptgpo01b/ReportServer?%2FFinancial%2FHISCI%20Best%20Practice%20Roster&amp;rs%3AFormat=XML&amp;rs%3ASnapshot%3Aisnull=True&amp;rc%3ASchema=True" Name="HISCI Best Practice Roster" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="HISCI_x0020_Best_x0020_Practice_x0020_Roster"><Tablix2 Textbox378="Company: --All--" Textbox49="Portfolio: --All--" Textbox375="Group: --All--" Textbox51="Member Status: Both" Textbox372="Division: --All--" Textbox52="Info. Type: --ALL--" Textbox366="Market: --All--" Textbox53="From: 09/02/2016" Textbox383="Class of Trade: --All--" Textbox54="To:  " Textbox385="Facility Type: --All--" /><Tablix32><Details_Collection><Details txtPharmacyGPOIDValue2="1" txtPharmacyGLNValue2="1" HealthIndustryNumber="1" txtPharmacyONameValue2="1" Name2="" Textbox17="Ship To" /></Details_Collection></Tablix32></Report>"#####;

    let _xml_value: Element = from_str(s).unwrap();
}
```

**Comment by e-oz** [comment]
Could you please try this file also, maybe issue is in BOM symbol?

[Report_test.2.xml.zip](https://github.com/serde-rs/xml/files/475092/Report_test.2.xml.zip)

**Comment by oli-obk** [comment]
zip files xD anyway, the xml is the same as the one you uploaded. It works just fine.

**Comment by oli-obk** [comment]
also the error location (line 1, col 43) makes no sense when looking at that piece of xml. Are you sure you uploaded the xml that fails on your system?

**Comment by e-oz** [comment]
Yes, I'm sure it's the same file - I checked it before sending it too you.
XML is the same, but zipped file have BOM symbols, they are invisible and can't not be copied in editor. That's why I attached file.
I wish I could say "works fine" too :)
Anyway, thanks for help.

**Comment by oli-obk** [comment]
never heard of BOM symbols... I'll try loading the file directly

**Comment by e-oz** [comment]
some badass invented them: https://en.wikipedia.org/wiki/Byte_order_mark

Github doesn't allow XML file to be attached, so I wrapped it into zip just to attach here.

**Comment by dtolnay** [comment]
This reproduces the issue.

``` rust
extern crate serde_xml;

use std::fs::File;
use std::io::Read;

#[test]
fn eoz() {
    let f = File::open("/path/to/Report_test.2.xml").unwrap().bytes();

    serde_xml::de::from_iter::<_, serde_xml::value::Element>(f).unwrap();
}
```

**Comment by oli-obk** [comment]
jup, I'm reproducing it right now, too. Must be due to the fact that serde-xml works on bytes and not on `char`s

**Comment by oli-obk** [comment]
we probably need to audit all text-based serde deserializers. Json probably bails with `ExpectedSomeValue`.

**Comment by e-oz** [comment]
I found a workaround:

```
from_str(xml_data.trim_matches('\u{FEFF}'))
```

It's legit and will not affect data without BOM.

**Comment by oli-obk** [comment]
hehe, don't worry, it's gonna be merged in a sec, and I'll probably release 0.9 right away

**Comment by e-oz** [comment]
Works great with 0.9.0, thank you very much!
